sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -917,4 +907,31 @@ export class StudentService extends RecordDataModelService<Student> {
       loadEagerRelations: false,
     });
   }
+
+  /**
+   * Creates a new CAS Supplier for a student.
+   * @param studentId student id.
+   * @param auditUser user.
+   * @param entityManager optional entity manager.
+   * @returns saved student.
+   **/
+  async createPendingCASSupplier(
+    studentId: number,
+    auditUser: User,
+    options?: { entityManager?: EntityManager },
+  ): Promise<Student> {
+    const studentRepo = options?.entityManager
+      ? options?.entityManager.getRepository(Student)
+      : this.repo;
+    const casSupplier = new CASSupplier();
+    casSupplier.supplierStatus = SupplierStatus.PendingSupplierVerification;
+    casSupplier.supplierStatusUpdatedOn = new Date();
+    casSupplier.isValid = false;
+    casSupplier.creator = auditUser;
+    casSupplier.student = { id: studentId } as Student;
+    const student = new Student();
+    student.id = studentId;
+    student.casSupplier = casSupplier;
```

Comment:
Need to add student audit data `modifier`.
If setting the `updated_at` the same data from `casSupplier.supplierStatusUpdatedOn` can be reused from a variable `now = new Date()`.